test/libsolidity/syntaxTests/inlineAssembly/blobhash_pre_cancun_warning.sol

```diff
@@ -0,0 +1,20 @@
+contract C {
+    function f() public pure returns (uint ret) {
+        assembly {
+            let blobhash := 1
+            ret := blobhash
+        }
+    }
+    function g() public pure returns (uint ret) {
+        assembly {
+            function blobhash() -> r {
+                r := 1000
+            }
+            ret := blobhash()
```

Comment:
A few more tests would be good. One for defining a function called ``blobhash`` for evm versions >= cancun, one for *using* ``blobhash`` without it being declared in ``< cancun``, one with ``blobhash`` being declared as variable and not a function, etc.

Reply:
Some of this is still missing. In particular the test for using `blobhash` on `<cancun`.

Reply:
Ah, I see you're working on that: https://github.com/ethereum/solidity/pull/14759#issuecomment-1889934553

Reply:
Done ;)